public/js/reducers/sources.js

```diff
@@ -224,15 +215,15 @@ function getSourceText(state: OuterState, id: string) {
 }
 
 function getSourceTabs(state: OuterState) {
-  return state.sources.tabs;
+  const tabs = state.sources.tabs;
+  return tabs
+    .filter(url => getSourceByURL(state, url))
+    .map(url => getSourceByURL(state, url));
```

Comment:
Sure. The map would return an object or null. Filter would remove the
undefined.

> Tab target URL

I think that's a good idea, but I'm not sure where it belongs. The
advantage is that we can always see which is the current debuggee URL

On Sun, Aug 28, 2016 at 5:56 PM Jaideep Bhoosreddy <notifications@github.com>
wrote:

> In public/js/reducers/sources.js
> <https://github.com/devtools-html/debugger.html/pull/603#discussion_r76539975>
> :
>
> > @@ -224,15 +215,15 @@ function getSourceText(state: OuterState, id: string) {
> >  }
> >
> >  function getSourceTabs(state: OuterState) {
> > -  return state.sources.tabs;
> > +  const tabs = state.sources.tabs;
> > +  return tabs
> > +    .filter(url => getSourceByURL(state, url))
> > +    .map(url => getSourceByURL(state, url));
>
> could you briefly tell me what would be the effect of adding a filter with
> an identity function to the result of map?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/devtools-html/debugger.html/pull/603/files/2879e218ebefe6b97861876a62646467a1bbbb1f#r76539975>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPiYtnxKUjsdNlpZfjvHZYs0gXQFWfsks5qkgPxgaJpZM4Juo30>
> .
>